### PR TITLE
Edit Button and Route Implemented

### PR DESCRIPTION
**Pull request by jksmall0631**
## Purpose
When viewing an existing reminder, the user can click on an “Edit” button where they can modify the title, notes, or date of the reminder.

When editing a note, there is a “Save” button that commits the changes to the server.
When the user hits the save button, they are returned to the original individual note path (ie: reminders/1).
Things to keep in mind:

What should your route look like to edit a reminder?

[Issue #11](https://github.com/turingschool-projects/1610-remember-1/issues/11)
## Approach

We started with creating a new nested route called 'edit'. We then wanted to add the edit button itself. We wanted to route to the card/edit and show all of the input fields and a save button once edit was clicked. We then saved all of the information on the save click and routed back to the original card route. We then wrote tests to show that everything works

We used plenty of Ember Docs for research

[Defining your routes](http://guides.emberjs.com/v2.11.0/routing/defining-your-routes/)
[Updating and Deleting Records](https://guides.emberjs.com/v2.11.0/models/creating-updating-and-deleting-records/)
[Mirage Route Handlers](http://www.ember-cli-mirage.com/docs/v0.2.0-beta.2/route-handlers/)
[How to Fix Mirage Problems](https://github.com/samselikoff/ember-cli-mirage/issues/187)

### Open Questions and Pre-Merge TODOs

- [x]  add new edit route
- [x]  make editing functionality work
- [x]  make submit button
- [x]  reroute back to single reminder on save
- [x]  add acceptance and integration tests on everything necessary

### Test coverage 

We wrote a few more acceptance tests and some integration tests for the edit route.

### Follow-up tasks

- [Issue #12](https://github.com/turingschool-projects/1610-remember-1/issues/12)

#### Before
screen shot 2017-02-11 at 7 23 15 pm

#### After
screen shot 2017-02-13 at 5 51 41 pm
screen shot 2017-02-13 at 5 51 59 pm
screen shot 2017-02-13 at 5 49 42 pm

### CODE REVIEW

File: tests/acceptance/reminder-list-test.js [hunk 1]
```diff
@@ -66,3 +66,30 @@ test('render warning message when nothing is added', function(assert) {
...
+test('edit and save fields', function(assert) {
+
+  visit('/');
+  click('.spec-reminder--new');
+  fillIn('.spec-input-title', 'Bananas');
+  fillIn('.spec-input-date', '12/23/16');
+  fillIn('.spec-textarea-notes', 'This shit is bananas, B.A.N.A.N.A.S.');
+  click('.spec-reminder-add--submit');
+  click('.spec-reminder--title');
+  click('.spec-reminder-card--editbtn');
```
Comment (Tman22) [inline tests/acceptance/reminder-list-test.js:79]: Can we fix this spacing

**Review by Tman22 — COMMENTED** [review]
Nice work team, let me know when you fix this and I'll merge it.

**Comment by jksmall0631** [comment]
@Tman22 @martensonbj Fixed the spacing

**Review by martensonbj — APPROVED** [review]
Niiiice

**Pull request merged**